### PR TITLE
Update API to Address Security Leak

### PR DESCRIPTION
**Pull request by eskildsvennungsen**
## Description:
This pull request updates our API to address a security vulnerability discovered in our previous implementation. The previous solution inadvertently exposed sensitive information, which could potentially be exploited. This update enhances our security measures to ensure the confidentiality and integrity of our data.

## Changes Made:
1. Replaced the exposed API endpoint with a more secure version.

## Background:
In light of the recent security review, it was identified that the previous implementation of our API had a significant vulnerability where sensitive data was inadvertently exposed. This update is crucial to protect our system from potential breaches and to safeguard the integrety of our quiz.

As Albert Einstein once said, 
> "In the middle of difficulty lies opportunity." 

This situation, while challenging, has given us the opportunity to improve our security posture and enhance our API's robustness.

## Testing:
- [ ] Verified that the new API endpoint is functioning correctly.
- [ ] Confirmed that all sensitive information is properly secured.

## Additional Notes:
Please review the changes and provide feedback. It's essential that we deploy this update as soon as possible to mitigate any risks associated with the previous implementation.

---

Let me know if you need any further adjustments!

### CODE REVIEW

**Comment by lopen** [comment]
### Comment on Merge Request

Hi Team,

Firstly, thank you for your hard work and dedication in implementing these new features. Your effort is highly appreciated, and it's clear a lot of thought and energy went into this update.

However, I have some concerns regarding the changes introduced in this merge request. After reviewing the code, it appears that some of the modifications may potentially break our existing codebase. Specifically, the new implementation seems to interfere with several core functions that are crucial for the current system's stability.

To ensure we maintain a smooth user experience and avoid any disruptions, I suggest we:
1. Conduct more extensive testing, especially focusing on edge cases and existing functionalities.
2. Review the integration points to ensure compatibility with the current codebase.
3. Collaborate on identifying and addressing any potential conflicts before merging.

I believe that with a bit more refinement, we can integrate these exciting new features without compromising our system's integrity. Let's work together to make this update a success!

Thank you for understanding, and I'm here to help with any needed adjustments.

Best,
Lopen

File: server/src/country.js [hunk 1]
```diff
@@ -2,22 +2,21 @@ const express = require('express');
...
 const cache = new NodeCache();
 const db = new Database('./countries.sqlite3');
 const apiSecret = process.env.API_SECRET;
 
-route.get('/distance', (req, res) => {
+route.get('/check', (req, res) => {
   try {
-    const from = getCountry(req.query.from);
-    const to = getCountry(req.query.to);
+    const target = getCountry(req.query.target);
```
Comment (lopen) [inline server/src/country.js:12]: name change is gonna affect frontend game logic

File: server/src/country.js [hunk 2]
```diff
@@ -2,22 +2,21 @@ const express = require('express');
...
+    const solution = getCountryOTD();
+
+    const correctGuess = target.name === solution.name;
 
-    const distance = calculateDistance(from, to);
+    const distance = calculateDistance(target, solution);
 
-    res.status(200).json({ a: from, b: to, distance: distance });
+    res.status(200).json({ correctGuess: correctGuess, distance: distance });
```
Comment (lopen) [inline server/src/country.js:19]: the frontend game logic uses the a: from field to populate the dynamic globe updates

Comment (eskildsvennungsen) [reply server/src/country.js:19]: The reason for this change is so that anyone cant go into inspect element and get the solution. I'm sorry this brings trouble in the front-end department, but it's absolutly vital that we protect the integrety of our quiz.

Comment (lopen) [reply server/src/country.js:19]: Could we make the correctGuess the data of the guessed country instead of a bool? I dont think thats gonna break the current implementation

File: server/src/country.js (hunk 1 above)
Comment (eskildsvennungsen) [reply server/src/country.js:12]: This should have been considered at an earlier stage. Actions will be taken so that errors like this does not repeat it self.

File: client/src/components/Input.jsx
Comment (eskildsvennungsen) [inline client/src/components/Input.jsx:1]: har en plugin som kødder med formatting så det ser ut som veldig mye er gjort, men har kun endret på linje 14, 16 og 18

File: client/src/components/Input.jsx [hunk 3]
```diff
@@ -3,30 +3,29 @@ import React from 'react';
...
-        <form className='opacity-100' method="post" onSubmit={handleSubmit}>
-            <label>
-                <input className='text-black text-center' name="newGuess" />
-            </label>
-            <hr />
-            <button type="submit">Submit</button>
-        </form>
-    )
-}
+    const arr = props.data.guesses.concat({ country: guess, distance: res.distance });
```
Comment (lopen) [inline client/src/components/Input.jsx:18]: guess her blir vel bare en bool?

Comment (eskildsvennungsen) [reply client/src/components/Input.jsx:18]: Nei, det er user inputten, ref. linje 14

Comment (lopen) [reply client/src/components/Input.jsx:18]: Ah, det er bare stringen som brukeren skriver i input feltet. Ikke faktisk data

Comment (eskildsvennungsen) [reply client/src/components/Input.jsx:18]: Ja, men innser nå at det er jo ikke noe sanity check på hva det er for no så denne løsningen fungerer ikke helt. Skal ordne opp i det

Comment (lopen) [reply client/src/components/Input.jsx:18]: sant, basically alt trenger å få en makeover. Men jeg synes fortsatt at check requesten kunne returna dataen til landet som ble gjetta

Comment (eskildsvennungsen) [reply client/src/components/Input.jsx]: Hvorfor? Du har jo allerede den dataen tilgjengelig siden man trenger det som input for api-endpointet

Comment (lopen) [reply client/src/components/Input.jsx]: Tar bare navn som input, er jo ikke noe mer data enn en string før det kommer til api
